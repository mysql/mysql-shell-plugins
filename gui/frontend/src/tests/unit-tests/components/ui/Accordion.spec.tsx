/*
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is designed to work with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have either included with
 * the program or referenced in the documentation.
 *
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

import { render } from "@testing-library/preact";
import { act } from "preact/test-utils";
import { describe, expect, it, vi } from "vitest";

import { Accordion } from "../../../../components/ui/Accordion/Accordion.js";
import { Codicon } from "../../../../components/ui/Codicon.js";
import { Assets } from "../../../../supplement/Assets.js";

const icon = Assets.documents.overviewPageIcon as string;

describe("Accordion component tests", (): void => {

    it("Test Accordion callbacks", async () => {
        const onSectionExpand = vi.fn();
        const onSectionAction = vi.fn();

        const { container, unmount } = render(
            <Accordion
                id="sidebar1"
                style={{
                    borderWidth: "1px",
                    borderStyle: "solid",
                }}
                caption="SIDEBAR TITLE"
                footer={"Lorem ipsum dolor sit amen, consenter"}
                singleExpand={true}
                onSectionExpand={onSectionExpand}
                onSectionAction={onSectionAction}

                sections={[
                    {
                        caption: "FIRST SECTION",
                        id: "first",
                        expanded: false,
                        content: [
                            <Accordion.Item id="item1" key="item1" caption="Item 1" />,
                            <Accordion.Item id="item2" key="item2" caption="Item 2" />,
                            <Accordion.Item id="item3" key="item3" caption="Item 3" />,
                        ],
                    },
                    {
                        caption: "SECOND SECTION",
                        actions: [
                            {
                                icon: Codicon.NewFile,
                                command: {
                                    title: "Add New Console",
                                    command: "addConsole",
                                    tooltip: "Add new console",
                                },
                            },
                        ],
                        id: "second",
                        content: [
                            <Accordion.Item
                                id="item5"
                                key="item5"
                                caption="Item 5"
                                picture={icon}
                            />,
                            <Accordion.Item
                                id="item6"
                                key="item6"
                                caption="Item 6"
                                picture={icon}
                            />,
                            <Accordion.Item
                                id="item7"
                                key="item7"
                                caption="Item 7"
                                picture={icon}
                            />,
                        ],
                    },
                    {
                        caption: "THIRD SECTION",
                        id: "third",
                        content: [
                            <Accordion.Item id="item9" key="item9" caption="Item 9" />,
                            <Accordion.Item id="item10" key="item10" caption="Item 10" />,
                            <Accordion.Item id="item11" key="item11" caption="Item 11" />,
                        ],
                    },
                ]}
            />,
        );

        const section1 = container.querySelector("#first") as HTMLElement;
        expect(section1).toBeDefined();

        const section2 = container.querySelector("#second") as HTMLElement;
        expect(section2).toBeDefined();

        const section3 = container.querySelector("#third");
        expect(section3).toBeDefined();

        expect(onSectionExpand).not.toHaveBeenCalled();

        await act(() => {
            const title = section1.getElementsByClassName("title")[0] as HTMLElement;
            title.click();
        });
        expect(onSectionExpand).toHaveBeenCalled();

        expect(onSectionAction).not.toHaveBeenCalled();
        const addConsoleButton = container.querySelector<HTMLButtonElement>("#addConsole");
        expect(addConsoleButton).toBeDefined();

        await act(() => {
            addConsoleButton?.click();
        });
        expect(onSectionAction).toHaveBeenCalled();

        unmount();
    });

    it("Test accordion output (Snapshot)", () => {
        const { container, unmount } = render(
            <Accordion
                id="sidebar1"
                style={{
                    borderWidth: "1px",
                    borderStyle: "solid",
                }}
                caption="SIDEBAR TITLE"
                footer={"Lorem ipsum dolor sit amen, consenter"}
                singleExpand={true}

                sections={[
                    {
                        caption: "FIRST SECTION",
                        id: "first",
                        expanded: false,
                        content: [
                            <Accordion.Item id="item1" key="item1" caption="Item 1" />,
                            <Accordion.Item id="item2" key="item2" caption="Item 2" />,
                            <Accordion.Item id="item3" key="item3" caption="Item 3" />,
                        ],
                    },
                    {
                        caption: "SECOND SECTION",
                        id: "second",
                        content: [
                            <Accordion.Item
                                id="item5"
                                key="item5"
                                caption="Item 5"
                                picture={icon}
                            />,
                            <Accordion.Item
                                id="item6"
                                key="item6"
                                caption="Item 6"
                                picture={icon}
                            />,
                            <Accordion.Item
                                id="item7"
                                key="item7"
                                caption="Item 7" picture={icon}
                            />,
                        ],
                    },
                    {
                        caption: "THIRD SECTION",
                        id: "third",
                        content: [
                            <Accordion.Item id="item9" key="item9" caption="Item 9" />,
                            <Accordion.Item id="item10" key="item10" caption="Item 10" />,
                            <Accordion.Item id="item11" key="item11" caption="Item 11" />,
                        ],
                    },
                ]}
            />,
        );
        expect(container).toMatchSnapshot();

        unmount();
    });
});
