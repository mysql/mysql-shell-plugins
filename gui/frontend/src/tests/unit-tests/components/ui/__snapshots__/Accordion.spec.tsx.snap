// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion component tests > Test accordion output (Snapshot) 1`] = `
<div>
  <div
    class="msg container fixedScrollbar accordion"
    id="sidebar1"
    style="flex-direction: column; justify-content: flex-start; border-width: 1px; border-style: solid;"
  >
    <div
      class="msg container fixedScrollbar actions"
      style="flex-direction: row; justify-content: flex-end; align-items: center;"
    >
      <label
        class="msg label title"
        data-tooltip="expand"
      >
        SIDEBAR TITLE
      </label>
    </div>
    <div
      class="msg splitContainer accordionContent resizing"
    >
      <div
        class="sashContainer"
      >
        <div
          class="msg divider horizontal sash disabled"
          id="firstSash"
          style="--thickness: 4px; top: 19px;"
        />
        <div
          class="msg divider horizontal sash disabled"
          id="secondSash"
          style="--thickness: 4px; top: 18px;"
        />
      </div>
      <div
        class="splitContentContainer"
      >
        <div
          class="msg container fixedScrollbar splitHost"
          id="firstHost"
          style="flex-direction: column; justify-content: stretch; height: 22px; top: 0px;"
        >
          <div
            class="msg container fixedScrollbar section"
            id="first"
            style="flex-direction: column; align-items: stretch;"
          >
            <div
              class="msg container fixedScrollbar title"
              style="flex-direction: row; align-items: center;"
              tabindex="0"
            >
              <div
                class="msg icon codicon codicon-chevron-right"
              />
              <label
                class="msg label"
                data-tooltip="expand"
              >
                FIRST SECTION
              </label>
              <div
                class="msg container fixedScrollbar actions"
                style="flex-direction: row; justify-content: flex-end; align-items: center;"
              />
            </div>
            <div
              class="msg container fixedScrollbar content"
              style="flex-direction: column;"
              tabindex="0"
            >
              <div
                class="msg container fixedScrollbar accordionItem"
                id="item1"
                style="flex-direction: row; align-items: center;"
                tabindex="0"
              >
                <label
                  class="msg label"
                  data-tooltip="expand"
                >
                  Item 1
                </label>
              </div>
              <div
                class="msg container fixedScrollbar accordionItem"
                id="item2"
                style="flex-direction: row; align-items: center;"
                tabindex="0"
              >
                <label
                  class="msg label"
                  data-tooltip="expand"
                >
                  Item 2
                </label>
              </div>
              <div
                class="msg container fixedScrollbar accordionItem"
                id="item3"
                style="flex-direction: row; align-items: center;"
                tabindex="0"
              >
                <label
                  class="msg label"
                  data-tooltip="expand"
                >
                  Item 3
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          class="msg container fixedScrollbar splitHost"
          id="secondHost"
          style="flex-direction: column; justify-content: stretch; top: 22px;"
        >
          <div
            class="msg container fixedScrollbar section expanded"
            id="second"
            style="flex-direction: column; align-items: stretch;"
          >
            <div
              class="msg container fixedScrollbar title"
              style="flex-direction: row; align-items: center;"
              tabindex="0"
            >
              <div
                class="msg icon codicon codicon-chevron-down"
              />
              <label
                class="msg label"
                data-tooltip="expand"
              >
                SECOND SECTION
              </label>
              <div
                class="msg container fixedScrollbar actions"
                style="flex-direction: row; justify-content: flex-end; align-items: center;"
              />
            </div>
            <div
              class="msg container fixedScrollbar content"
              style="flex-direction: column;"
              tabindex="0"
            >
              <div
                class="msg container fixedScrollbar accordionItem"
                id="item5"
                picture="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='UTF-8'%20standalone='no'?%3e%3c!DOCTYPE%20svg%20PUBLIC%20'-//W3C//DTD%20SVG%201.1//EN'%20'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'%3e%3csvg%20width='100%25'%20height='100%25'%20viewBox='0%200%2016%2016'%20version='1.1'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%20xml:space='preserve'%20xmlns:serif='http://www.serif.com/'%20style='fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;'%3e%3cg%20id='overviewPage'%20transform='matrix(1,0,0,1,1.46549e-14,1)'%3e%3cg%20transform='matrix(1.14286,0,0,1.14286,-1.14286,-2.28572)'%3e%3cpath%20d='M14.5,2L1.5,2L1,2.5L1,13.762L1.5,14.262L14.5,14.262L15,13.762L15,2.5L14.5,2ZM14.125,13.375L1.875,13.375L1.875,2.875L14.125,2.875L14.125,13.375Z'%20style='fill-rule:nonzero;'/%3e%3c/g%3e%3cg%20id='symbol-structure'%20transform='matrix(1.13557,0,0,1.13557,-1.11994,-3.23989)'%3e%3cg%20transform='matrix(1,0,0,1,0,3.52247)'%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3cg%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,9.31536,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
                style="flex-direction: row; align-items: center;"
                tabindex="0"
              >
                <img
                  class="msg image"
                  height="20px"
                  src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='UTF-8'%20standalone='no'?%3e%3c!DOCTYPE%20svg%20PUBLIC%20'-//W3C//DTD%20SVG%201.1//EN'%20'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'%3e%3csvg%20width='100%25'%20height='100%25'%20viewBox='0%200%2016%2016'%20version='1.1'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%20xml:space='preserve'%20xmlns:serif='http://www.serif.com/'%20style='fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;'%3e%3cg%20id='overviewPage'%20transform='matrix(1,0,0,1,1.46549e-14,1)'%3e%3cg%20transform='matrix(1.14286,0,0,1.14286,-1.14286,-2.28572)'%3e%3cpath%20d='M14.5,2L1.5,2L1,2.5L1,13.762L1.5,14.262L14.5,14.262L15,13.762L15,2.5L14.5,2ZM14.125,13.375L1.875,13.375L1.875,2.875L14.125,2.875L14.125,13.375Z'%20style='fill-rule:nonzero;'/%3e%3c/g%3e%3cg%20id='symbol-structure'%20transform='matrix(1.13557,0,0,1.13557,-1.11994,-3.23989)'%3e%3cg%20transform='matrix(1,0,0,1,0,3.52247)'%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3cg%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,9.31536,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
                  width="20px"
                />
                <label
                  class="msg label"
                  data-tooltip="expand"
                >
                  Item 5
                </label>
              </div>
              <div
                class="msg container fixedScrollbar accordionItem"
                id="item6"
                picture="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='UTF-8'%20standalone='no'?%3e%3c!DOCTYPE%20svg%20PUBLIC%20'-//W3C//DTD%20SVG%201.1//EN'%20'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'%3e%3csvg%20width='100%25'%20height='100%25'%20viewBox='0%200%2016%2016'%20version='1.1'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%20xml:space='preserve'%20xmlns:serif='http://www.serif.com/'%20style='fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;'%3e%3cg%20id='overviewPage'%20transform='matrix(1,0,0,1,1.46549e-14,1)'%3e%3cg%20transform='matrix(1.14286,0,0,1.14286,-1.14286,-2.28572)'%3e%3cpath%20d='M14.5,2L1.5,2L1,2.5L1,13.762L1.5,14.262L14.5,14.262L15,13.762L15,2.5L14.5,2ZM14.125,13.375L1.875,13.375L1.875,2.875L14.125,2.875L14.125,13.375Z'%20style='fill-rule:nonzero;'/%3e%3c/g%3e%3cg%20id='symbol-structure'%20transform='matrix(1.13557,0,0,1.13557,-1.11994,-3.23989)'%3e%3cg%20transform='matrix(1,0,0,1,0,3.52247)'%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3cg%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,9.31536,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
                style="flex-direction: row; align-items: center;"
                tabindex="0"
              >
                <img
                  class="msg image"
                  height="20px"
                  src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='UTF-8'%20standalone='no'?%3e%3c!DOCTYPE%20svg%20PUBLIC%20'-//W3C//DTD%20SVG%201.1//EN'%20'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'%3e%3csvg%20width='100%25'%20height='100%25'%20viewBox='0%200%2016%2016'%20version='1.1'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%20xml:space='preserve'%20xmlns:serif='http://www.serif.com/'%20style='fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;'%3e%3cg%20id='overviewPage'%20transform='matrix(1,0,0,1,1.46549e-14,1)'%3e%3cg%20transform='matrix(1.14286,0,0,1.14286,-1.14286,-2.28572)'%3e%3cpath%20d='M14.5,2L1.5,2L1,2.5L1,13.762L1.5,14.262L14.5,14.262L15,13.762L15,2.5L14.5,2ZM14.125,13.375L1.875,13.375L1.875,2.875L14.125,2.875L14.125,13.375Z'%20style='fill-rule:nonzero;'/%3e%3c/g%3e%3cg%20id='symbol-structure'%20transform='matrix(1.13557,0,0,1.13557,-1.11994,-3.23989)'%3e%3cg%20transform='matrix(1,0,0,1,0,3.52247)'%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3cg%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,9.31536,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
                  width="20px"
                />
                <label
                  class="msg label"
                  data-tooltip="expand"
                >
                  Item 6
                </label>
              </div>
              <div
                class="msg container fixedScrollbar accordionItem"
                id="item7"
                picture="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='UTF-8'%20standalone='no'?%3e%3c!DOCTYPE%20svg%20PUBLIC%20'-//W3C//DTD%20SVG%201.1//EN'%20'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'%3e%3csvg%20width='100%25'%20height='100%25'%20viewBox='0%200%2016%2016'%20version='1.1'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%20xml:space='preserve'%20xmlns:serif='http://www.serif.com/'%20style='fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;'%3e%3cg%20id='overviewPage'%20transform='matrix(1,0,0,1,1.46549e-14,1)'%3e%3cg%20transform='matrix(1.14286,0,0,1.14286,-1.14286,-2.28572)'%3e%3cpath%20d='M14.5,2L1.5,2L1,2.5L1,13.762L1.5,14.262L14.5,14.262L15,13.762L15,2.5L14.5,2ZM14.125,13.375L1.875,13.375L1.875,2.875L14.125,2.875L14.125,13.375Z'%20style='fill-rule:nonzero;'/%3e%3c/g%3e%3cg%20id='symbol-structure'%20transform='matrix(1.13557,0,0,1.13557,-1.11994,-3.23989)'%3e%3cg%20transform='matrix(1,0,0,1,0,3.52247)'%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3cg%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,9.31536,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
                style="flex-direction: row; align-items: center;"
                tabindex="0"
              >
                <img
                  class="msg image"
                  height="20px"
                  src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='UTF-8'%20standalone='no'?%3e%3c!DOCTYPE%20svg%20PUBLIC%20'-//W3C//DTD%20SVG%201.1//EN'%20'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'%3e%3csvg%20width='100%25'%20height='100%25'%20viewBox='0%200%2016%2016'%20version='1.1'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%20xml:space='preserve'%20xmlns:serif='http://www.serif.com/'%20style='fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;'%3e%3cg%20id='overviewPage'%20transform='matrix(1,0,0,1,1.46549e-14,1)'%3e%3cg%20transform='matrix(1.14286,0,0,1.14286,-1.14286,-2.28572)'%3e%3cpath%20d='M14.5,2L1.5,2L1,2.5L1,13.762L1.5,14.262L14.5,14.262L15,13.762L15,2.5L14.5,2ZM14.125,13.375L1.875,13.375L1.875,2.875L14.125,2.875L14.125,13.375Z'%20style='fill-rule:nonzero;'/%3e%3c/g%3e%3cg%20id='symbol-structure'%20transform='matrix(1.13557,0,0,1.13557,-1.11994,-3.23989)'%3e%3cg%20transform='matrix(1,0,0,1,0,3.52247)'%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3cg%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,2.27042,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,5.79288,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3cg%20transform='matrix(0.704493,0,0,0.659639,9.31536,-1.32241)'%3e%3cpath%20d='M1,10L2,9L4.384,9L5.384,10L5.384,12.005L4.384,13.005L2,13.005L1,12.005L1,10ZM4.384,10L2,10L2,12.005L4.384,12.005L4.384,10Z'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
                  width="20px"
                />
                <label
                  class="msg label"
                  data-tooltip="expand"
                >
                  Item 7
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          class="msg container fixedScrollbar splitHost"
          id="thirdHost"
          style="flex-direction: column; justify-content: stretch; top: 21px;"
        >
          <div
            class="msg container fixedScrollbar section expanded"
            id="third"
            style="flex-direction: column; align-items: stretch;"
          >
            <div
              class="msg container fixedScrollbar title"
              style="flex-direction: row; align-items: center;"
              tabindex="0"
            >
              <div
                class="msg icon codicon codicon-chevron-down"
              />
              <label
                class="msg label"
                data-tooltip="expand"
              >
                THIRD SECTION
              </label>
              <div
                class="msg container fixedScrollbar actions"
                style="flex-direction: row; justify-content: flex-end; align-items: center;"
              />
            </div>
            <div
              class="msg container fixedScrollbar content"
              style="flex-direction: column;"
              tabindex="0"
            >
              <div
                class="msg container fixedScrollbar accordionItem"
                id="item9"
                style="flex-direction: row; align-items: center;"
                tabindex="0"
              >
                <label
                  class="msg label"
                  data-tooltip="expand"
                >
                  Item 9
                </label>
              </div>
              <div
                class="msg container fixedScrollbar accordionItem"
                id="item10"
                style="flex-direction: row; align-items: center;"
                tabindex="0"
              >
                <label
                  class="msg label"
                  data-tooltip="expand"
                >
                  Item 10
                </label>
              </div>
              <div
                class="msg container fixedScrollbar accordionItem"
                id="item11"
                style="flex-direction: row; align-items: center;"
                tabindex="0"
              >
                <label
                  class="msg label"
                  data-tooltip="expand"
                >
                  Item 11
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <label
      class="msg label footer"
      data-tooltip="expand"
    >
      Lorem ipsum dolor sit amen, consenter
    </label>
  </div>
</div>
`;
